components/shared/BalanceInput.test.ts

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { mount, createLocalVue } from '@vue/test-utils'
+import Buefy from 'buefy'
+import { vi } from 'vitest'
+
+import BalanceInput from '@/components/shared/BalanceInput.vue'
+
+const localVue = createLocalVue()
+localVue.use(Buefy)
+
+test('BalanceInput.vue', async () => {
+  vi.useFakeTimers()
+
+  const wrapper = mount(BalanceInput, {
+    attachTo: document.body,
+    mocks: {
+      $t: () => '',
+      $config: {
+        prefix: 'rmrk',
+      },
+      $store: {
+        getters: {
+          currentUrlPrefix: 'rmrk',
+        },
+      },
+    },
+    localVue,
+  })
+  const input = wrapper.find('[data-testid="balance-input"]')
+  const options = wrapper
+    .find('[data-testid="balance-input-select"]')
+    .findAll('option')
+
+  // test focus
+  await (wrapper.vm as any).focusInput()
+  expect(input.element).toBe(document.activeElement)
+
+  // check html5validity
+  await wrapper.setProps({ min: 10, max: 100 })
+  await input.setValue('0')
+  expect(await (wrapper.vm as any).checkValidity()).toBe(false)
+  await input.setValue('50')
+  expect(await (wrapper.vm as any).checkValidity()).toBe(true)
+  await input.setValue('900')
+  expect(await (wrapper.vm as any).checkValidity()).toBe(false)
+
+  // set value from props
+  await wrapper.setProps({ value: 3 })
+  expect((input.element as HTMLInputElement).value).toBe('3')
+
+  // check value
+  await input.setValue('1')
+  vi.runAllTimers()
+  expect(wrapper.emitted().input).toBeTruthy()
+  expect(wrapper.emitted().input?.at(-1)).toStrictEqual(['1000000000000', '1'])
```

Comment:
are these value chain-dependent?

Reply:
currently mocked to `rmrk` only https://github.com/kodadot/nft-gallery/pull/3663/files#diff-a69b9ffc5ec6df32d1010499df90ddc122682e54f89e5bf7bafdccbedcdd5ab5R23
since `bsx` also uses 12 as decimals, the output will same